spec/services/shopify_refunder_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'spec_helper'
+
+require_relative '../../lib/active_merchant/billing/gateways/shopify.rb'
+
+describe ShopifyRefunder do
+  let(:transaction_id) { '0xDEADBEEF' }
+  let(:options) { { reason: 'Actual reason', order_id: transaction_id } }
+  let(:transaction_amount) { 1 }
+  let(:credited_money_in_cents) { 100 }
+  let(:transaction) { double(:transaction, amount: transaction_amount, id: transaction_id) }
+
+  before do
+    allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
+  end
+
+  context '.initialize' do
+    subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
+    it 'successfully does it\'s thing' do
+      expect(subject).to be_truthy
+    end
+  end
+
+  context '.perform' do
+    let(:pos_refund) { double('refund', errors: []) }
+
+    before do
+      allow(ShopifyAPI::Refund).to receive(:create).and_return(pos_refund)
+    end
+
+    context 'when the shopify transaction is not found' do
+      before do
+        allow(ShopifyAPI::Transaction).to receive(:find).and_return(nil)
+      end
+
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
+      it 'throws an error' do
+        cause = ->{ subject.perform }
+        expect(&cause).to raise_error(ActiveMerchant::Billing::ShopifyGateway::TransactionNotFoundError)
+      end
+    end
+
+    context 'with a full refund' do
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
+      it 'performs a refund on shopify' do
+        expect(ShopifyAPI::Refund).to receive(:create)
```

Comment:
```
        refund = refunder_interface.create(order_id: order_id,
                                           shipping: { amount: 0 },
                                           note: refund_reason,
                                           notify: false,
                                           restock: false,
                                           transactions: [{
                                             parent_id: transaction.id,
                                             amount: amount_to_dollars(credited_money),
                                             gateway: 'shopify-payments',
                                             kind: 'refund'
                                           }])
```

That's a lot of argument, you would recommend having all of them written in the spec itself?